### ISSUE TITLE
[Client/Server] Invalid data from the server

### ISSUE THREAD

**Issue by a-rmz** [a-rmz/JEasyCrypto#9]
## Description
If JSON parsing fails on the server side, the request id etc. are not valid data from client.

## Steps to Reproduce the Problem
  1. 🔨
  2. 🔧
  3. 💥

**Comment by avengerpb** [comment]
How to reproduce this bugs ?

**Comment by a-rmz** [comment]
@anttijuu can you help me with this one please?

**Comment by anttijuu** [comment]
Yes I can. So when you look at the server code, at line 52 onwards:

```
		JSONObject root = (JSONObject) new JSONParser().parse(receivedData);
	
		id = ((Long) root.get("id")).longValue(); // id of the operation, for async operations.
		operation = (String) root.get("operation"); // request/response
```
If the parse() call fails, id and operation is never extracted from the client request. That means, when creating the response to the client...:

```
} catch (ParseException e) {
	e.printStackTrace();
	response = createResponse(operation, id, new EasyCryptoAPI.Result(EasyCryptoAPI.ResultCode.EError, e.getLocalizedMessage()));
```

...the id and operation parameters are null. Thus, the client can not see from the response, which of the requests it sent failed. Client could send several requests quickly and they could be executed by the server in different order, and different requests might take different time to fulfil. So client cannot be sure that responses are received in the same order than the requests were sent.

This can be tricky to solve, since if parse fails, it means the JSON is not valid for some reason. The JSON data from client might contain the request id, but maybe it doesn't even have that. 

You could try to investigate the string and look for the id without using the JSON parser to extract the request id in this case. Or return some more informative response to the client, telling it that the JSON it sent was garbage and the client probably has a bug and/or the data was corrupted during transmission from client to the server. Or something.
